### ISSUE TITLE
Form validation

### ISSUE THREAD

**Issue by m4rk055** [blindnet-io/product-management#303]
Form validation is inconsistent across our projects, even within a single project. We need to standardize it in [Product Design Principles](https://docs.google.com/document/d/1drrvedXnoaflBLRfQw9qVh5MRxV_3rH-sAf47DdjKD4).

According to many articles:
https://designmodo.com/ux-form-validation/
https://cxl.com/blog/form-validation/
https://blog.prototypr.io/ux-best-practices-of-form-validation-ddb8a0df14fd
the best way to do the validation is immediately after a form field is filled in, rather than after submitting the form.
It reduces friction and user frustration.

I propose the following:
- When a form is initialized, no field has an error
- After a field is focused for the first time, wait until a user finishes filling it in
- After the field is unfocused for the first time, validate it.
  - Validation passed: no decoration
  - Validation failed: error indicator - red border with an error message*
- After the field is focused for the second time and onward, field validation runs on every value change so a user knows immediately when the value is acceptable
- Submitting the form validates all fields.
  - Validation passed: send a request to server
  - Validation failed: put an error indicator to every field with a bad value. From this point onward, all fields behave as if they were "focused" once

*Error message should be under the field rather than in tooltip. If there are many fields with errors, tooltip becomes congested and hard to follow.

![image](https://user-images.githubusercontent.com/71560916/137328056-53831778-cbd9-4a02-a329-0de067fe314c.png)

**Comment by Clementinev** [comment]
Close as resolved in https://github.com/blindnet-io/product-management/issues/472

**Comment by Clementinev** [comment]
I added it in the following PR: https://github.com/blindnet-io/product-management/pull/797
